dataclients/kubernetes/clusterclient.go

```diff
@@ -400,6 +405,32 @@ func (c *clusterClient) loadServices() (map[definitions.ResourceID]*service, err
 	return result, nil
 }
 
+func (c *clusterClient) loadSecrets() (map[definitions.ResourceID]*secret, error) {
+	var secrets secretList
+	if err := c.getJSON(c.secretsURI, &secrets); err != nil {
+		log.Debugf("requesting all secrets failed: %v", err)
+		return nil, err
+	}
+
+	log.Debugf("all secrets received: %d", len(secrets.Items))
+	result := make(map[definitions.ResourceID]*secret)
+	var hasInvalidSecret bool
+	for _, secret := range secrets.Items {
+		if secret == nil || secret.Meta == nil {
+			hasInvalidSecret = true
+			continue
+		}
+
+		result[secret.Meta.ToResourceID()] = secret
+	}
+
+	if hasInvalidSecret {
+		log.Errorf("Invalid secret resource detected.")
+	}
```

Comment:
IMO this does not help to debug the problem, we either need to log (maybe warn) each invalid secret namespace+name or silently ignore them

Reply:
The secret is invalid because it's either `nil` or  it does not have metadata (name+namespace) so we cannot warn with that info. Updating so silently ignore.